src/utils/payout.ts

```diff
@@ -0,0 +1,74 @@
+import { abi as ERC20 } from '@openzeppelin/contracts/build/contracts/ERC20.json'
+import EthersAdapter from '@gnosis.pm/safe-ethers-lib'
+import Safe from '@gnosis.pm/safe-core-sdk'
+import { SafeTransactionDataPartial } from '@gnosis.pm/safe-core-sdk-types'
+import * as ethers from 'ethers'
+
+export type PayoutParams = {
+	network?: number;
+	from: string;
+	amount: string;
+	token: string; // pass 0x00 to send ether
+	to: string;
+  execute?: boolean;
+  account?: number;
+  gwei?: number;
+}
+
+export const providerKeyByChainId = new Map<number, string>([
+  [1, process.env.RPC1_ALCHEMY_KEY as string],
+  [5, process.env.RPC5_ALCHEMY_KEY as string], // goerli - safe compatable
+])
+
+export const payout = async (args: PayoutParams) => {
+  // collect baseline info
+  const tokenAddress = ethers.utils.getAddress(args.token)
```

Comment:
shouldn't we make sure the token address is whitelisted ?

Reply:
That's a nice detail. This is to make sure we don't waste gas I presume?

Reply:
correct. good addition @zgorizzo69

Reply:
do you mean something specific @zgorizzo69 when you use whitelisted? i was just thinking - balance, and allowed is set to cover the amount. is there anything else?